ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh

```diff
@@ -42,17 +63,42 @@ if [[ -z "$CLUSTER_ID" ]]; then
 fi
 
 echo "Deleting cluster-id: ${CLUSTER_ID}"
+start_time=$(date +"%s")
 rosa delete cluster -c "${CLUSTER_ID}" -y
-while rosa describe cluster -c "${CLUSTER_ID}" ; do
-  sleep 60
+while true; do
+  CLUSTER_STATE=$(rosa describe cluster -c "${CLUSTER_ID}" -o json 2>/dev/null | jq -r '.state' || true)
+  echo "Cluster state: ${CLUSTER_STATE}"
+  current_time=$(date +"%s")
+  if (( "${current_time}" - "${start_time}" >= "${DESTROY_TIMEOUT}" )); then
+    echo "error: Cluster not deleted after ${DESTROY_TIMEOUT}"
+    exit 1
+  else
+    if [[ "${CLUSTER_STATE}" == "error" ]]; then
+      echo "Cluster ${CLUSTER_ID} is on error state and wont be deleted."
+      exit 1
+    elif [[ "${CLUSTER_STATE}" == "" ]]; then
+      end_time=$(date +"%s")
+      echo "Cluster destroyed after $(( ${end_time} - ${start_time} )) seconds"
+      record_cluster "timers" "destroy" $(( "${end_time}" - "${start_time}" ))
+      break
+    else
+      echo "Cluster ${CLUSTER_ID} is on ${CLUSTER_STATE} state, waiting 60 seconds for the next check"
+      sleep 60
+    fi
+  fi
 done
 
 if [[ "$STS" == "true" ]]; then
+  start_time=$(date +"%s")
   echo "Deleting operator roles"
   rosa delete operator-roles -c "${CLUSTER_ID}" -y -m auto
-  
+
   echo "Deleting oidc-provider"
   rosa delete oidc-provider -c "${CLUSTER_ID}" -y -m auto
+
+  end_time=$(date +"%s")
+  record_cluster "timers" "sts_destroy" $(( "${end_time}" - "${start_time}" ))
```

Comment:
@morenod Should this be recorded irrespective of STS cluster or not? 
Our current PerfScale tests all create STS clusters only and there is a plan to move to STS by default soon. However, should we ever need to time a non-sts cluster, the record_cluster won't be invoked. Just calling it out so we make informed decision.

Reply:
if not using sts, there is no operator-roles or oidc-providers to remove, so no, this section requires the if and the var of STS to be set. if we move, in the future, to only use sts, and we remove the conditional, this block will be a must